File: src/main/java/org/railwaystations/rsapi/adapter/in/web/controller/PhotographersController.java
package org.railwaystations.rsapi.adapter.in.web.controller;

import lombok.RequiredArgsConstructor;
import org.railwaystations.rsapi.adapter.in.web.api.PhotographersApi;
import org.railwaystations.rsapi.core.ports.in.LoadPhotographersUseCase;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.RestController;

@RestController
@RequiredArgsConstructor
public class PhotographersController implements PhotographersApi {

    private final LoadPhotographersUseCase loadPhotographersUseCase;

    @Override
    public ResponseEntity<Object> photographersGet(String country) {
        return ResponseEntity.ok(loadPhotographersUseCase.getPhotographersPhotocountMap(country));
    }

}

File: adapter/src/main/kotlin/org/railwaystations/rsapi/adapter/web/controller/PhotographersController.kt
package org.railwaystations.rsapi.adapter.web.controller

import org.railwaystations.rsapi.adapter.web.api.PhotographersApi
import org.railwaystations.rsapi.core.ports.inbound.LoadPhotographersUseCase
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.RestController

@RestController
class PhotographersController(private val loadPhotographersUseCase: LoadPhotographersUseCase) : PhotographersApi {

    override fun getPhotographers(country: String?): ResponseEntity<Any> {
        return ResponseEntity.ok(loadPhotographersUseCase.getPhotographersPhotocountMap(country))
    }

}
